blocks/bricks/recurrent.py

```diff
@@ -177,9 +177,16 @@ def scan_function(*args):
                 args = list(args)
                 arg_names = (list(sequences_given) + list(states_given) +
                              list(contexts_given))
-                kwargs = dict(zip(arg_names, args))
+                kwargs = OrderedDict(zip(arg_names, args))
                 kwargs.update(rest_kwargs)
-                return application_function(brick, **kwargs)
+                outputs = application_function(brick, **kwargs)
```

Comment:
I have just got a crazy idea: what if instead we do
```python
getattr(brick, application_function.__name__)(**dict_union(kwargs, iterate=False))
```
here? That makes the inner graph to look exactly like if the user built it manually!